src/Microsoft.Azure.WebJobs.ServiceBus/MessagingProvider.cs

```diff
@@ -37,46 +32,21 @@ public MessagingProvider(ServiceBusConfiguration config)
         }
 
         /// <summary>
-        /// Creates a <see cref="MessagingFactory"/> for the specified ServiceBus entity.
-        /// </summary>
-        /// <param name="entityPath">The ServiceBus entity to create a <see cref="MessagingFactory"/> for.</param>
-        /// <param name="connectionStringName">Optional connection string name indicating the connection string to use.
-        /// If null, the default connection string on the <see cref="ServiceBusConfiguration"/> will be used.</param>
-        /// <returns>A <see cref="MessagingFactory"/>.</returns>
-        /// <remarks>For performance reasons, factories are cached per connection string.</remarks>
-        public virtual MessagingFactory CreateMessagingFactory(string entityPath, string connectionStringName = null)
-        {
-            if (string.IsNullOrEmpty(entityPath))
-            {
-                throw new ArgumentNullException("entityPath");
-            }
-
-            string connectionString = GetConnectionString(connectionStringName);
-
-            // We cache messaging factories per entity/connection, in accordance with ServiceBus
-            // performance guidelines.
-            // https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-performance-improvements
-            string cacheKey = $"{entityPath}-{connectionString}";
-            var messagingFactory = _messagingFactoryCache.GetOrAdd(cacheKey, (p) =>
-            {
-                return MessagingFactory.CreateFromConnectionString(connectionString);
-            });
-
-            return messagingFactory;
-        }
-
-        /// <summary>
         /// Creates a <see cref="MessageProcessor"/> for the specified ServiceBus entity.
         /// </summary>
         /// <param name="entityPath">The ServiceBus entity to create a <see cref="MessageProcessor"/> for.</param>
         /// <returns>The <see cref="MessageProcessor"/>.</returns>
-        public virtual MessageProcessor CreateMessageProcessor(string entityPath)
+        public virtual MessageProcessor CreateMessageProcessor(string entityPath, string connectionString)
```

Comment:
`CreateMessageProcessor` and `CreateMessageReceiver` each create a new `MessageReceiver` for the same entityPath/connStr now. It looks like that means that each trigger will now create 2 `MessageReceiver`s. One will be used to receive messages and one will be used by the `MessageProcessor` to mark messages as complete. 

This wasn't an issue previously b/c `CompleteAsync` lived on the `BrokeredMessage` but now it looks like it moved to the `MessageReceiver`. 

Maybe this is by design -- but it jumped out at me.

Reply:
I added  MessageReceiver cache like we have for MessagingFactory in v1:
https://github.com/Azure/azure-webjobs-sdk/blob/e915ee574136207cab3b01cac7139bf09b013ace/src/Microsoft.Azure.WebJobs.ServiceBus/MessagingProvider.cs#L24